File: 260-SingleNumberIii/260-SingleNumberIii.js
// Last updated: 3/7/2026, 10:02:43 PM
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var singleNumber = function(nums) {
   let obj = {};
for(let i=0; i<nums.length; i++){
    if(obj[nums[i]]=== undefined){
        obj[nums[i]] =1;
    }
    else{
   obj[nums[i]] ++;
    }
}
let arr =[]
for(let key in obj){
    if(obj[key] === 1){
     arr.push(Number(key))
    }
}
 return arr;
};

File: 260-SingleNumberIii/260-SingleNumberIii.cpp
// Last updated: 8/18/2026, 11:53:31 AM
class Solution {
public:
    vector<int> singleNumber(vector<int>& nums) {
        unordered_map<int,int> freq;
        vector<int> res;

        for(int i=0; i<nums.size(); i++){
            freq[nums[i]]++;
        }

        for(auto it : freq){
            if(it.second == 1){
                res.push_back(it.first);
            }
        }
        return res;
    }
};